test/object-store/sync/app.cpp

```diff
@@ -1948,7 +1949,31 @@ constexpr size_t minus_25_percent(size_t val)
     return val * .75 - 10;
 }
 
+static void set_app_config_defaults(app::App::Config& app_config,
+                                    const std::shared_ptr<app::GenericNetworkTransport>& transport)
+{
+    if (!app_config.transport)
+        app_config.transport = transport;
+    if (app_config.platform.empty())
+        app_config.platform = "Object Store Test Platform";
+    if (app_config.platform_version.empty())
+        app_config.platform_version = "Object Store Test Platform Version";
+    if (app_config.sdk_version.empty())
+        app_config.sdk_version = "SDK Version";
+    if (app_config.app_id.empty())
+        app_config.app_id = "app_id";
+    if (!app_config.local_app_version)
+        app_config.local_app_version.emplace("A Local App Version");
+}
+
 TEST_CASE("app: sync integration", "[sync][app]") {
+    auto logger = std::make_unique<util::StderrLogger>();
+#if TEST_ENABLE_SYNC_LOGGING
+    logger->set_level_threshold(util::Logger::Level::all);
```

Comment:
You can just use this define https://github.com/realm/realm-core/blob/master/test/object-store/util/test_file.hpp#L228 to setup the log level if you've included test_file.hpp already.